base/parse.jl

```diff
@@ -38,14 +38,42 @@ julia> parse(Complex{Float64}, "3.2e-1 + 4.5im")
 parse(T::Type, str; base = Int)
 parse(::Type{Union{}}, slurp...; kwargs...) = error("cannot parse a value as Union{}")
 
-function parse(::Type{T}, c::AbstractChar; base::Integer = 10) where T<:Integer
-    a::Int = (base <= 36 ? 10 : 36)
-    2 <= base <= 62 || throw(ArgumentError("invalid base: base must be 2 ≤ base ≤ 62, got $base"))
-    d = '0' <= c <= '9' ? c-'0'    :
-        'A' <= c <= 'Z' ? c-'A'+10 :
-        'a' <= c <= 'z' ? c-'a'+a  : throw(ArgumentError("invalid digit: $(repr(c))"))
-    d < base || throw(ArgumentError("invalid base $base digit $(repr(c))"))
-    convert(T, d)
+@noinline _invalid_base(base) = throw(ArgumentError("invalid base: base must be 2 ≤ base ≤ 62, got $base"))
+@noinline _invalid_digit(base, char) = throw(ArgumentError("invalid base $base digit $(repr(char))"))
+
+function parse_char(
+        ::Type{T},
+        c::AbstractChar,
+        base::Integer,
+        throw::Bool
+    ) where {T <: Integer}
+    a::UInt8 = (base <= 36 ? 10 : 36)
+    if !(2 <= base <= 62)
+        throw ? _invalid_base(base) : return nothing
+    end
+    base = base % UInt8
+    codepoint = c > 'z' ? 0xff : UInt8(c)
+    d = if UInt8('0') ≤ codepoint ≤ UInt8('9')
+        codepoint - UInt8('0')
+    elseif UInt8('A') ≤ codepoint ≤ UInt8('Z')
+        codepoint - UInt8('A') + 0x0a # + 10 for the 10 digits
```

Comment:
```suggestion
        codepoint - UInt8('A') + UInt8(10) # + 10 for the 10 digits
```
Given that the comment is referring to "10" and the other UInt8s around here are expressed as conversion from more semantically clear types.